tests/ops/op_math/test_change_op_basis.py

```diff
@@ -0,0 +1,348 @@
+    """Variant of qml.RX that claims to not have `adjoint` or a matrix defined."""
+
+    has_matrix = False
+    has_adjoint = False
+    has_decomposition = False
+    has_diagonalizing_gates = False
+
+
+class TestInitialization:  # pylint:disable=too-many-public-methods
+    """Test the initialization."""
+
+    def test_init_change_op_basis_op(self):
+        """Test the initialization of a ChangeOpBasis operator."""
+        change_op_basis_op = ChangeOpBasis(qml.PauliX(wires=0), qml.RZ(0.23, wires="a"))
+
+        assert change_op_basis_op.wires == Wires((0, "a"))
+        assert change_op_basis_op.num_wires == 2
+        assert change_op_basis_op.name == "ChangeOpBasis"
+
+        assert change_op_basis_op.data == (0.23,)
+        assert change_op_basis_op.parameters == [0.23]
+        assert change_op_basis_op.num_params == 1
+
+    def test_hash(self):
+        """Testing some situations for the hash property."""
+        # test not the same hash if different order
+        op1 = qml.change_op_basis(qml.PauliX("a"), qml.PauliY("a"), qml.PauliX(1))
+        op2 = qml.change_op_basis(qml.PauliY("a"), qml.PauliX("a"), qml.PauliX(1))
+        assert op1.hash != op2.hash
+
+    def test_batch_size(self):
+        """Test that batch size returns the batch size of a base operation if it is batched."""
+        x = qml.numpy.array([1.0, 2.0, 3.0])
+        change_op_basis_op = change_op_basis(qml.PauliX(0), qml.RX(x, wires=0))
+        assert change_op_basis_op.batch_size == 3
+
+    def test_batch_size_None(self):
+        """Test that the batch size is none if no factors have batching."""
+        change_op_basis_op = change_op_basis(qml.PauliX(0), qml.RX(1.0, wires=0))
+        assert change_op_basis_op.batch_size is None
+
+    @pytest.mark.parametrize(
+        "factors",
+        (
+            [qml.PauliX(wires=0), qml.Hamiltonian([0.5], [qml.PauliX(wires=1)])],
+            [qml.PauliX(wires=0), qml.RZ(0.612, "r")],
+            [
+                qml.Hamiltonian([-0.3], [qml.PauliZ(wires=1)]),
+                qml.Hamiltonian([0.5], [qml.PauliX(wires=1)]),
+            ],
+            [MyOp(3.1, 0), qml.CNOT([0, 2])],
+        ),
+    )
+    def test_has_adjoint_true_always(self, factors):
+        """Test that a change_op_basis of operators that have `has_adjoint=True`
+        has `has_adjoint=True` as well."""
+
+        change_op_basis_op = change_op_basis(*factors)
+        assert change_op_basis_op.has_adjoint is True
+
+    @pytest.mark.parametrize(
+        "factors",
+        (
+            [qml.PauliX(wires=0), qml.Hamiltonian([0.5], [qml.PauliX(wires=1)])],
+            [qml.PauliX(wires=0), qml.RZ(0.612, "r")],
+            [
+                qml.Hamiltonian([-0.3], [qml.PauliZ(wires=1)]),
+                qml.Hamiltonian([0.5], [qml.PauliX(wires=1)]),
+            ],
+            [MyOp(3.1, 0), qml.CNOT([0, 2])],
+        ),
+    )
+    def test_has_decomposition_true_always(self, factors):
```

Comment:
So if we allow `Hamiltonian` to be an operand in this structure, the decomposition isn't always defined, because it wouldn't even be a unitary operation anymore.